### DISCUSSION TITLE
Apply custom-theme.yaml to generatePDF

### DISCUSSION THREAD

**Discussion by tobfranz** [docToolchain/docToolchain#842] [Q&A] [answered]
Hey,
at first, i want to say thank you for all your great work here.
Now my question:
How can i generate a pdf with a custom layout with the docToolchain wrapper?

In older versions (1.y.z) it worked when my custom-theme.yaml directory was doc/pdfTheme.
Now i upgraded to the dtcw and the theme was not applied. Then i found https://doctoolchain.org/docToolchain/v2.0.x/015_tasks/03_task_generatePDF.html and tried it, but it does not work. I inserted :pdf-stylesdir: ../pdfTheme with different directories to my main adoc and my config.adoc file. Then i tried https://doctoolchain.org/docToolchain/v2.0.x/015_tasks/03_task_copy_themes.html, but it also not worked without and with and with other different ways. I also tried to add pdfThemeDir to my doctoolchainConfig.groovy.

So i connected to the docker container and toked a look at /home/dtcuser/docToolchain/scripts/AsciiDocBasics.gradle. (https://github.com/docToolchain/docToolchain/blob/ae91937e22f1fd3d7e6467713d14650d76dcc3d4/scripts/AsciiDocBasics.gradle) There i found out that pdf-stylesdir and pdf-style are comment out under attributes.

```
...
   //tag::AsciidoctorTask[]
  
   // common settings for asciidoctor
   // this is needed for PDF generation with plantUML
   tasks.withType(org.asciidoctor.gradle.AsciidoctorTask) { docTask ->
      config.taskInputsDirs.each {
          inputs.dir new File(docDir, it)
      }
      config.taskInputsFiles.each {
          inputs.file new File(docDir, it)
      }

    // configure source and output files and folders
    outputDir = file(targetDir)
    sourceDir = file(srcDir)
    attributes  (
    //        'pdf-stylesdir': "${pdfThemeDir}@",
    //           'pdf-style': "custom@",
    //           'pdf-fontsdir': "${pdfThemeDir}/fonts@",
            'source-highlighter': 'highlightjs@',
            'imagesdir': 'images@',
            'toc': 'left@',
            'icons': 'font@',
            'javaVersion'         : "$javaVersion",
            'currentDate'         : "$currentDate",
            'allow-uri-read'      : 'true@',
            'sectanchors'         : 'true@',
            'targetDir'           : targetDir,
            'docDir'              : docDir,
            'projectRootDir'      : "${new File(docDir).canonicalPath}@",
    )
...
```

In docToolchain v1.4.0 the attributes section looks different. pdf-stylesdir and pdf-style are comment in.

In the file from the docker container i commented the attributes in and run the generatePDF task from the project folder. The pdf looked like expected. The pdfThemeDir was included in my doctoolchainConfig.groovy.

Did i miss something in the config or did i add the custeme-theme.yaml wrong or is this a bug?

Maybe someone can explain me the right way to add a custeme-theme.

Best regards

T. Franz

**Comment by rdmueller** [comment]
thanx for the detailed problem report.
As far as I remember, the default pdf-styles shipped with docToolchain caused problems and thus, these settings where removed.

your first approach, to add the pdf-styles location to the document should work. Let me give this a try.

**Comment by rdmueller** [comment, marked as answer]
so, I tried to write down the correct way on how to configure the pdfTheme here in the tutorial:

http://doctoolchain.org/docToolchain/v2.0.x/020_tutorial/030_generateHTML.html#_pdf_style

as always: feedback is welcome!

**Reply by tobfranz** [reply]
Hey,
many thanks for the quick help. I think ``:pdf-style: custom``did the trick. For me there was no need for the copyThemes task.
My xml examples look a bit different, but that's a new topic i have to look for.

Best regards

T. Franz

**Reply by rdmueller** [reply]
the copyThemes helps to get a starting point for the theme.

let's talk about the xml examples - in another thread...
